pkgs/io_file/lib/src/windows_file_system.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:ffi';
+import 'dart:io' as io;
+import 'package:ffi/ffi.dart';
+import 'package:win32/win32.dart' as win32;
+
+import 'file_system.dart';
+
+String _formatMessage(int errorCode) {
+  final buffer = win32.wsalloc(1024);
+  try {
+    final result = win32.FormatMessage(
+      win32.FORMAT_MESSAGE_FROM_SYSTEM | win32.FORMAT_MESSAGE_IGNORE_INSERTS,
+      nullptr,
+      errorCode,
+      0, // default language
+      buffer,
+      1024,
+      nullptr,
+    );
+    if (result == 0) {
+      return '';
+    } else {
+      return buffer.toDartString();
+    }
+  } finally {
+    win32.free(buffer);
+  }
+}
+
+Exception _getError(int errorCode, String message, String path) {
+  final osError = io.OSError(_formatMessage(errorCode), errorCode);
+
+  switch (errorCode) {
+    case win32.ERROR_ACCESS_DENIED:
+    case win32.ERROR_CURRENT_DIRECTORY:
+    case win32.ERROR_WRITE_PROTECT:
+    case win32.ERROR_BAD_LENGTH:
+    case win32.ERROR_SHARING_VIOLATION:
+    case win32.ERROR_LOCK_VIOLATION:
+    case win32.ERROR_NETWORK_ACCESS_DENIED:
+    case win32.ERROR_DRIVE_LOCKED:
+      return io.PathAccessException(path, osError, message);
+    case win32.ERROR_FILE_EXISTS:
+    case win32.ERROR_ALREADY_EXISTS:
+      return io.PathExistsException(path, osError, message);
+    case win32.ERROR_FILE_NOT_FOUND:
+    case win32.ERROR_PATH_NOT_FOUND:
+    case win32.ERROR_INVALID_DRIVE:
+    case win32.ERROR_INVALID_NAME:
+    case win32.ERROR_NO_MORE_FILES:
+    case win32.ERROR_BAD_NETPATH:
+    case win32.ERROR_BAD_NET_NAME:
+    case win32.ERROR_BAD_PATHNAME:
+      return io.PathNotFoundException(path, osError, message);
+    default:
+      return io.FileSystemException(message, path, osError);
+  }
+}
+
+/// A [FileSystem] implementation for Windows systems.
+base class WindowsFileSystem extends FileSystem {
+  @override
+  void rename(String oldPath, String newPath) {
+    // Calling `GetLastError` for the first time causes the `GetLastError`
+    // symbol to be loaded, which resets `GetLastError`. So make a harmless
+    // call before the value is needed.
+    win32.GetLastError();
```

Comment:
Do we need to do this before every call that might record an error? Or once, before the first win32 call?

If the 2nd, perhaps do this as part of static initialization of the win32 filesystem.

Reply:
I'm hoping to fix this in the SDK so I can just remove this code.